identity/src/test/java/com/android/identity/SessionEncryptionInteractionTest.java

```diff
@@ -0,0 +1,107 @@
+        PublicKey eDeviceKeyPublic = eDeviceKeyPair.getPublic();
+        PrivateKey eDeviceKeyPrivate = eDeviceKeyPair.getPrivate();
+
+        byte[] encodedSessionTranscript = new byte[] {1, 2, 3};
+
+        SessionEncryptionReader sessionEncryptionReader = new SessionEncryptionReader(
+                eReaderKeyPrivate, eReaderKeyPublic, eDeviceKeyPublic, encodedSessionTranscript);
+        sessionEncryptionReader.setSendSessionEstablishment(false);
+        SessionEncryptionDevice sessionEncryptionHolder = new SessionEncryptionDevice(
+                eDeviceKeyPrivate, eReaderKeyPublic, encodedSessionTranscript);
+
+        // have reader generate sessionEstablishment and check holder decryption of sessionEstablishment
+        byte[] mdocRequest = sessionEncryptionReader.encryptMessageToDevice(
+                Util.fromHex(TestVectors.ISO_18013_5_ANNEX_D_DEVICE_REQUEST),
+                OptionalLong.empty());
+        Pair<byte[], OptionalLong> result = sessionEncryptionHolder.decryptMessageFromReader(
+                mdocRequest);
+        Assert.assertFalse(result.second.isPresent());
+        Assert.assertArrayEquals(
+                Util.fromHex(TestVectors.ISO_18013_5_ANNEX_D_DEVICE_REQUEST),
+                result.first);
+
+        // have holder generate deviceResponse and check reader decryption of deviceResponse
+        byte[] deviceResponse = sessionEncryptionHolder.encryptMessageToReader(
+                Util.fromHex(TestVectors.ISO_18013_5_ANNEX_D_DEVICE_RESPONSE),
+                OptionalLong.empty());
+        result = sessionEncryptionReader.decryptMessageFromDevice(deviceResponse);
+        Assert.assertFalse(result.second.isPresent());
+        Assert.assertArrayEquals(
+                Util.fromHex(TestVectors.ISO_18013_5_ANNEX_D_DEVICE_RESPONSE),
+                result.first);
+
+        // have reader generate message with status and ensure that holder can read it
+        byte[] readerSessionTermination = sessionEncryptionReader.encryptMessageToDevice(null,
+                OptionalLong.of(Constants.SESSION_DATA_STATUS_SESSION_TERMINATION));
+        result = sessionEncryptionHolder.decryptMessageFromReader(readerSessionTermination);
+        Assert.assertEquals(Constants.SESSION_DATA_STATUS_SESSION_TERMINATION,
+                result.second.getAsLong());
+        Assert.assertNull(result.first);
+
+        // have holder generate message with status and ensure that reader can read it
+        byte[] holderSessionTermination = sessionEncryptionHolder.encryptMessageToReader(null,
+                OptionalLong.of(Constants.SESSION_DATA_STATUS_SESSION_TERMINATION));
+        result = sessionEncryptionReader.decryptMessageFromDevice(holderSessionTermination);
+        Assert.assertEquals(Constants.SESSION_DATA_STATUS_SESSION_TERMINATION,
+                result.second.getAsLong());
+        Assert.assertNull(result.first);
+    }
+
+    @Test
+    @SmallTest
+    public void testAllCurves() {
```

Comment:
It's generally cleaner to have one test per curve. Then you can also add `@Ignore` for the 25519 and 448 curves.